api/cron/takedown_unpublished_models.py

```diff
@@ -18,11 +20,44 @@
 from utils.deployer import ModelDeployer  # # noqa isort:skip
 
 
+def parse_args():
+    parser = argparse.ArgumentParser()
+    parser.add_argument(
+        "--task_code",
+        type=str,
+        default="",
+        help=("Task code to filter models by " "when choosing to delete "),
+    )
+    args = parser.parse_args()
+
+    return args
+
+
 def main():
+    args = parse_args()
     m = ModelModel()
-    unpublished_models = m.getByPublishStatus(publish_status=False)
-    for model in unpublished_models:
-        if model.deployment_status == DeploymentStatusEnum.deployed:
+
+    # Takedown all unpublished models
+    models_to_takedown = set(m.getByPublishStatus(publish_status=False))
+
+    if args.task_code.strip() != "":
+        ops = input(
+            f"Take down all models associated with task code {args.task_code}? [y/n]"
```

Comment:
What do you think about changing this to:

"Take down all model endpoints associated with task code {args.task_code}, including published models? [y/n]"